packages/@react-spectrum/combobox/test/ComboBox.test.js

```diff
@@ -2024,23 +2023,45 @@ describe('ComboBox', function () {
       );
 
       let button = getByRole('button');
+      let combobox = getByRole('combobox');
 
-      await waitFor(() => expect(load).toHaveBeenCalledTimes(1));
+      expect(load).toHaveBeenCalledTimes(1);
       expect(onOpenChange).toHaveBeenCalledTimes(0);
       expect(onLoadMore).toHaveBeenCalledTimes(0);
 
+      // open menu
+      triggerPress(button);
+      // use async act to resolve initial load
       await act(async () => {
-        triggerPress(button);
-        jest.runAllTimers();
+        // advance to open state from Transition
+        jest.advanceTimersToNextTimer();
       });
-
       let listbox = getByRole('listbox');
       expect(listbox).toBeVisible();
+      jest.spyOn(listbox, 'clientHeight', 'get').mockImplementation(() => 100);
+      // update size, virtualizer raf kicks in
+      act(() => {jest.advanceTimersToNextTimer();});
+      // onLoadMore queued by previous timer, run it now
+      act(() => {jest.advanceTimersToNextTimer();});
 
       expect(onOpenChange).toHaveBeenCalledTimes(1);
       expect(onOpenChange).toHaveBeenCalledWith(true, 'manual');
       expect(onLoadMore).toHaveBeenCalledTimes(1);
-      await waitFor(() => expect(load).toHaveBeenCalledTimes(1));
+      expect(load).toHaveBeenCalledTimes(1);
+
+      // close menu
+      act(() => {combobox.blur();});
+      // raf from virtualizer relayout
+      act(() => {jest.advanceTimersToNextTimer();});
+      // previous act wraps up onExiting
+      // raf
+      act(() => {jest.advanceTimersToNextTimer();});
+      // raf
+      act(() => {jest.advanceTimersToNextTimer();});
+      // exited
+      act(() => {jest.advanceTimersToNextTimer();});
```

Comment:
Why are we wanting to timers so specifically? Verse `jest.runAllTimers()`

Reply:
each timer causes a state change, all state changes must happen within an act, once the act is done, the state changes are resolved. in this case, the state changes must be resolved separately because they influence what happens to the next timer.

there are still instances where you'd want to potentially advance multiple timers at once, which is what happens in the majority of our tests